### PR TITLE
Krulevskiy Maksim (ІМ-34) exam

### PR DESCRIPTION
**Pull request by MaksimkaKrul**
Done day.js, filter.js, month.js, sample.js, skip.js, distinct.js, longest.js, sum.js, take.js and words.js.
Not a lot of steps as commits, but comments show my steps and work by chronological order!
Hoping for a high score! :D
Glory to Allah!

### CODE REVIEW

File: Tasks/sample.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
+//Just one line function
 
-const SAMPLE = (arr) => {
-  arr = arr[Math.floor(Math.random() * arr.length)];
-  return arr;
-};
+'use strict'
 
-module.exports = SAMPLE;
+const randomArrayElem = (arr) => arr[Math.floor(Math.random() * arr.length)];
```
Comment (tshemsedinov) [inline Tasks/sample.js:9]: Maybe you can decompose it into two functions according to single responsibility principle?

File: Tasks/day.js [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
-  let i;
-  for (i = 0; i < D.length; i++) {
-    if (s.startsWith(D[i].toLowerCase())) {
-      return i + 1;
-    }
+const getDayIndex = (dayName) => {
+  let dayNameList = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];  
+
+  for (const [index, day] of dayNameList.entries()) {
+    if (dayName.indexOf(day.toLowerCase()) == 0) return index + 1;
```
Comment (tshemsedinov) [inline Tasks/day.js:15]: Why not strict comparison?

File: Tasks/filter.js [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
+//instead of changing arr, now we create filteredArray and filter arr to complete the task
+'use strict'
 
-Filter = (T, t) => {
-  remove = [];
-  for (C of T) {
-    x = T.indexOf(C);
-    if (typeof T[x] !== t) {
-      remove.unshift(x);
+const filterByType = (arr, type) => {
```
Comment (tshemsedinov) [inline Tasks/filter.js:9]: Can be done in one loop

File: Tasks/month.js [hunk 4]
```diff
@@ -1,26 +1,29 @@
...
+    'jan','feb','mar',
+    'apr','may','jun',
+    'jul','aug','sep',
+    'oct','nov','dec',
+  ];
+  
+  let lowerMonthName = monthName.toLowerCase();
+
+  for (const [index, month] of monthList.entries()) {
+    if (lowerMonthName.indexOf(month) == 0) return index + 1;
```
Comment (tshemsedinov) [inline Tasks/month.js:22]: Array.prototype.includes

File: Tasks/words.js [hunk 5]
```diff
@@ -1,41 +1,14 @@
...
-        }
-      }
-    }
+//For beginning I fixed bad naming and deleted unnececary flag
+//The flag thing is still here and will fix in next commit
+//Deleted all of the flags, and instead of it just count with .split
+//Deleted stupid \n after if statement 
+'use strict'
+const countWordsInArray = (inputString) => {
+  if (inputString == ''){
```
Comment (tshemsedinov) [inline Tasks/words.js:8]: Better single line

File: Tasks/sample.js (hunk 1 above)
Comment (MaksimkaKrul) [reply Tasks/sample.js]: Done! Now it decomposed into two functions.

File: Tasks/day.js (hunk 2 above)
Comment (MaksimkaKrul) [reply Tasks/day.js]: Fixed that too, such a small, but such an important detail!

File: Tasks/month.js (hunk 4 above)
Comment (MaksimkaKrul) [reply Tasks/month.js]: Fixed too! Looks even better and easier with .includes .

File: Tasks/words.js (hunk 5 above)
Comment (MaksimkaKrul) [reply Tasks/words.js]: Fixed too! Guess my eyes lost they sharpness at the end of exam 😄 .

File: Tasks/day.js [hunk 6]
```diff
@@ -1,15 +1,20 @@
...
+'use strict'
 
-_parse_day_ = (s) => {
-  let i;
-  for (i = 0; i < D.length; i++) {
-    if (s.startsWith(D[i].toLowerCase())) {
-      return i + 1;
-    }
+const getDayIndex = (dayName) => {
+  let dayNameList = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
```
[the remark this answers is not in the document]
Comment (tshemsedinov) [reply Tasks/day.js:12]: Move out of func

[the remark this answers is not in the document]
Comment (MaksimkaKrul) [reply Tasks/day.js:12]: > Move out of func

so can I ask why will we move out it? faster because if have list in function it will initialize everytime?

**Pull request closed without merging**